core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java

```diff
@@ -1523,5 +1529,31 @@ private void validateStringCharacterLimit() {
     }
   }
 
-
+  /**
+   * This method validates the DETAIL_QUERY_BATCH_SIZE. If some invalid input is set, we use the
+   * default value for this property
+   */
+  private void validateDetailQueryBatchSize() {
+    String batchSizeString =
+        carbonProperties.getProperty(CarbonCommonConstants.DETAIL_QUERY_BATCH_SIZE);
+    if (batchSizeString == null) {
+      carbonProperties.setProperty(DETAIL_QUERY_BATCH_SIZE,
+          Integer.toString(CarbonCommonConstants.DETAIL_QUERY_BATCH_SIZE_DEFAULT));
+      LOGGER.info(
+          "Using default value for carbon.detail.batch.size " + DETAIL_QUERY_BATCH_SIZE_DEFAULT);
+    } else {
+      int batchSize;
+      try {
+        batchSize = Integer.parseInt(batchSizeString);
+        if (batchSize < 100 || batchSize > 1000) {
+          throw new NumberFormatException();
```

Comment:
Why you are throwing exception if batch size value is not in the range of min max then use default value and add a log

Reply:
I was throwing exception because even if batchSize is some invalid character then it will get caught and I am using the default value and adding a log in the Catch part.So both the scenarios are covered here min max one and invalid character also.

Reply:
yes, add a log there instead of exception.

Reply:
Done.